spec/config/configuration_spec.rb

```diff
@@ -279,4 +282,40 @@ def consume
         }
       )
   end
+
+  it 'should override global configurations' do
+    described_class.configure do
+      consumers.bulk_import_id_generator(-> { 'global' })
+      consumers.replace_associations true
+
+      consumer do
+        class_name 'MyConfigConsumer'
+        schema 'blah'
+        topic 'blah'
+        group_id 'myconsumerid'
+        bulk_import_id_generator(-> { 'consumer' })
+        replace_associations false
+      end
+
+      consumer do
+        class_name 'MyConfigConsumer2'
+        schema 'blah'
+        topic 'blah'
+        group_id 'myconsumerid'
+      end
+    end
+
+    consumers = described_class.config.consumers
+    expect(consumers.replace_associations).to eq(true)
+    expect(consumers.bulk_import_id_generator.call).to eq('global')
+
+    custom = MyConfigConsumer.config
+    expect(custom[:replace_associations]).to eq(false)
+    expect(custom[:bulk_import_id_generator].call).to eq('consumer')
+
+    default = MyConfigConsumer2.config
+    expect(default[:replace_associations]).to eq(true)
+    expect(default[:bulk_import_id_generator].call).to eq('global')
```

Comment:
This doesn't actually test that the generator is called when expected (i.e. during the batch updates). Ideally you'd make this change in e.g. the mass updater spec.